cmake/linker/ld/target_base.cmake

```diff
@@ -10,10 +10,12 @@ macro(toolchain_ld_base)
 
   # TOOLCHAIN_LD_FLAGS comes from compiler/gcc/target.cmake
   # LINKERFLAGPREFIX comes from linker/ld/target.cmake
-  zephyr_ld_options(
-    -no-pie
-    ${TOOLCHAIN_LD_FLAGS}
-  )
+  if(${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "xcc-clang")
+    zephyr_ld_options(-fno-pie)
+  else()
+    zephyr_ld_options(-no-pie)
```

Comment:
`-fno-pie` and `-no-pie` are two different things.

The linker is often invoked through the compiler which makes things even more "interesting".

I have no idea whether this code is correct but for sure it requires a ton of comments as this is a super confusing area.

Reply:
Thanks @marc-hb, I'll double check it, marked PR as do not merge for now.